src/Compilers/Server/VBCSCompilerTests/NamedPipeClientConnectionHostTests.cs

```diff
@@ -57,13 +57,13 @@ public async Task CallAfterComplete()
         }
 
         [ConditionalFact(typeof(WindowsOrLinuxOnly), Reason = "https://github.com/dotnet/runtime/issues/40301")]
-        public void EndListenCancelsIncompleteTask()
+        public async Task EndListenCancelsIncompleteTask()
         {
             _host.BeginListening();
             var task = _host.GetNextClientConnectionAsync();
             _host.EndListening();
 
-            Assert.ThrowsAsync<OperationCanceledException>(() => task);
+            await Assert.ThrowsAsync<TaskCanceledException>(async () => await task);
```

Comment:
Seems like the lambda shouldn't need to change?

Reply:
The lambda wasn't actually doing anything before hence the exception wasn't ever checked.